frontend/src/entities/group/detail/group-detail-user.actions.tsx

```diff
@@ -0,0 +1,121 @@
+    const addableUsers = allUsers.filter((user) => !groupUsernames.includes(user.username!));
+    const [addUserModalVisible, setAddUserModalVisible] = useState(false);
+
+    return (
+        <SpaceBetween direction='horizontal' size='xs'>
+            <AddGroupUserModal dispatch={dispatch} setVisible={setAddUserModalVisible} visible={addUserModalVisible} addableUsers={addableUsers} groupName={groupName}/>
+            <Button onClick={() => dispatch(getGroupUsers(groupName))} ariaLabel={'Refresh groups list'}>
+                <Icon name='refresh'/>
+            </Button>
+            {GroupDetailUserActionsButton(navigate, dispatch, props)}
+            <Button
+                variant='primary'
+                onClick={() => setAddUserModalVisible(true)}
+            >
+                Add User
+            </Button>
+        </SpaceBetween>
+    );
+}
+
+function GroupDetailUserActionsButton (navigate: NavigateFunction, dispatch: Dispatch, props?: any) {
+    const selectedUser: IGroupUser = props?.selectedItems[0];
+    const items = [];
+    if (selectedUser) {
+        items.push({
+            text: 'Remove from Group',
+            id: 'removeFromGroup',
+        });
+    }
+
+    const [modalState, setModalState] = React.useState<Partial<ModalProps>>({
+        visible: false,
+        confirmText: 'Confirm',
+        dismissText: 'Cancel',
+    });
+
+    return (
+        <ButtonDropdown
+            items={items}
+            variant='primary'
+            disabled={!selectedUser}
+            onItemClick={(e) => GroupDetailUserActionHandler(e, dispatch, modalState as ModalProps, setModalState, selectedUser)}
+        >
+            Actions
+        </ButtonDropdown>
+    );
+}
+
+const GroupDetailUserActionHandler = async (
+    e: any,
+    dispatch:  ThunkDispatch<any, any, Action>,
+    modalState: ModalProps,
+    setModalState: (state: Partial<ModalProps>) => void,
+    selectedUser: IGroupUser
+) => {
+    const notificationService = NotificationService(dispatch);
+
+    switch (e.detail.id) {
+        case 'removeFromGroup':
+            dispatch(
+                setDeleteModal({
+                    resourceName: 'Group User',
+                    resourceType: 'groupUser',
+                    postConfirm: () => dispatch(getGroupUsers(selectedUser.group)),
+                    onConfirm: async () =>
+                        dispatch(removeGroupUser(selectedUser)).then((result) => {
+                            setModalState({
+                                ...modalState,
+                                visible: false,
+                            });
+                            notificationService.showActionNotification(
+                                'remove group user',
+                                `User ${selectedUser.user} removed from ${selectedUser.group}.`,
+                                result
+                            );
+                        }),
```

Comment:
The typing of this is so wacky I don't even have an immediate suggestion on how to fix it. Part of it is just how `onConfirm` was typed to begin with.